### PR TITLE
MET-1007 export menu item is available only to ADMIN and ROLE_METADATA_CURATOR

### PR DESCRIPTION
**Pull request by kuceram**
@musketyr Is this what you've been expected? :point_left: 

I don't check this on server side, so you can basically hack that but I understood this to be fine...

### CODE REVIEW

File: ModelCatalogueCorePlugin/grails-app/assets/javascripts/modelcatalogue/core/ui/bs/actions.coffee [hunk 1]
```diff
@@ -241,7 +241,7 @@ angular.module('mc.core.ui.bs.actions', ['mc.util.ui.actions']).config ['actions
   ]
 
   actionsProvider.registerActionInRoles 'export', [actionsProvider.ROLE_LIST_ACTION, actionsProvider.ROLE_ITEM_ACTION, actionsProvider.ROLE_NAVIGATION, actionsProvider.ROLE_LIST_HEADER_ACTION], ['$scope', 'security', ($scope, security)->
-    return undefined unless security.hasRole('VIEWER')
+    return undefined unless security.hasRole('ADMIN') || security.hasRole('ROLE_METADATA_CURATOR')
```
Comment (musketyr) [inline ModelCatalogueCorePlugin/grails-app/assets/javascripts/modelcatalogue/core/ui/bs/actions.coffee]: In front-end you should just use the role `CURATOR`. 

see https://github.com/MetadataRegistry/ModelCataloguePlugin/blob/2.x/ModelCatalogueCorePluginTestApp/grails-app/views/index.gsp#L102

**Comment by musketyr** [comment]
just wonder if we should check this on the backend as well. it sound as a good practice to me.

File: ModelCatalogueCorePlugin/grails-app/assets/javascripts/modelcatalogue/core/ui/bs/actions.coffee (hunk 1 above)
Comment (kuceram) [inline ModelCatalogueCorePlugin/grails-app/assets/javascripts/modelcatalogue/core/ui/bs/actions.coffee]: Now I see where these aggregated role comes from. I was wondering as I could not find them in backend. :sunglasses:

**Comment by kuceram** [comment]
Well, for me it is totally good practice to build non-hackable apps. :-) But this seemed to me as a bigger task which might break some other functionality of the app. So I would recommend to either create new task for it (as we should go through all API end-points and secure them accordingly) or we can also neglect it as the app is used by closed group of people - this is more business decision. :-)

My estimate to this issue is 2 - 3 hours. What you thinks? @musketyr @amilward

**Comment by musketyr** [comment]
it depends on what may happen in the worst case. reports consume lot of resources so it's good to protect this by more restricted rule than just plain `VIEWER`. user not logged in cannot use any of the actions anyway so only use case is a `VIEWER` who wants to hack and create the new report herself which is not very likely and can be tracked down easily so I would leave it as it is at the moment

**Pull request merged**